main.tex

```diff
@@ -905,6 +904,58 @@ \subsection{The effect of population size}\label{sec:population_size}
     \label{fig:correlation_coefficients}
 \end{figure}
 
+\section{Discussion}
+
+Training strategies to excel
+at the Moran process leads to the evolution of cooperation, but only with like
+individuals in the case of TF1 and TF2. This may have significant implications
+for human social interactions such as the evolution of ingroup/outgroup mechanisms
+and other sometimes costly rituals that reinforce group behavior.
+
+While TF1 and TF2 are competent invaders, the best invaders
+in the study do not appear to employ strict handshakes, and are generally
+cooperative strategies. TF3, which does not use a handshake, is a better invader
+than TF1 and TF2 but not as good of a resistor. Nevertheless it was the result
+of the same kind of training procceses and is a better combined invader-resistor
+than the invaders that were trained previously to maximize payout.
+
+The strategies trained to maximize payoff in head-to-head matches are generally
+cooperative and are effective invaders.
+Combined with the fact that handshaking strategies are stronger resisters,
+this suggests that while maximizing individual payoff can lead to the evolution
+of cooperation, these strategies are not the most evolutionarily stable
+in the long run. A strategy with a handshaking mechanism is still capable of
+invading and is more resistant to subsequent invasions. Moreover, the
+best resistor of the payoff maximally trained strategies (EvolvedLookerUp1_1_1),
+which always defects if the opponent defects in the first round, is effectively
+employing a one-shot handshake of C. These insights also suggest that a strategy
+aware of the population distribution could choose to become a handshaker at
+a critical threshold and use a different strategy for invasion when in the
+minority. However this information was not available to our strategies.
+
+We did not attempt other objective functions that may serve to select for both
+invasion and resistance better than training at a starting population of
+$(N/2, N/2)$. Nevertheless our results suggest that there is not much room for
+improvement. Any handshake more sophisticated than C necessarily involves
+a defection. (A handshake consisting of a longer sequence of cooperations is
+effectively a grudger.) For TF3 or EvolvedLookerUp1_1_1 to become better resistors
+they need a longer or more strict handshake. But if this handshake involves
+a defection then likely the invasion ability is diminished for $N > 2$: the top
```

Comment:
`then it is likely that the invasion`? (I don't know, as is read funky to me, I might be wrong)